snuba/query/parser/expressions.py

```diff
@@ -159,19 +205,29 @@ def parse_aggregation(
 
     columns_expr = [parse_expression(column) for column in columns if column]
 
-    if isinstance(parsed_expression, str):
-        # Simple aggregation with snuba syntax ["count", ["c1", "c2"]]
-        return FunctionCall(alias, parsed_expression, tuple(columns_expr))
-    elif (
-        # Simple Clickhouse expression with no snuba syntax
-        # ["ifNull(count(somthing), something)", None, None]
-        isinstance(parsed_expression, (FunctionCall, CurriedFunctionCall))
-        and not columns_expr
-    ):
-        return replace(parsed_expression, alias=alias)
-    elif isinstance(parsed_expression, FunctionCall) and columns_expr:
-        # Mix of clickhouse syntax and snuba syntax that generates a CurriedFunction
-        # ["f(a)", "b", None]
-        return CurriedFunctionCall(alias, parsed_expression, tuple(columns_expr),)
+    matched = re.fullmatch(function_name_regex, aggregation_function)
+
+    if matched is not None:
+        return FunctionCall(alias, aggregation_function, tuple(columns_expr))
+
     else:
```

Comment:
Since the previous `if` has only one statement inside, you can remove the else block:
```
if matched is not None:
    return FuncitonCall...

expression_tree = minimal_clickhouse_grammar.parse(aggregation_function)
.. all the rest ...